First Multichain API e2e Test

adonesky1 opened the pull request:
WIP

adonesky1 committed:
wip

adonesky1 commented:
@metamaskbot update-policies

adonesky1 committed:
allow-script

adonesky1 commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

adonesky1 committed:
dedupe

adonesky1 commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

adonesky1 committed:
more wip

adonesky1 committed:
lint

adonesky1 committed:
prolly need data-test-ids

adonesky1 committed:
fix

adonesky1 committed:
a first test working

adonesky1 committed:
lint

adonesky1 committed:
lint

jiexi committed:
use extensionId from setup

jiexi committed:
Merge branch 'ad/multichain-api-e2e-test' of github.com:MetaMask/metamask-extension into ad/multichain-api-e2e-test

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
yarn dedupe

jiexi committed:
yarn

jiexi commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

jiexi committed:
yarn dedupe

jiexi committed:
lint

jiexi commented:
@SocketSecurity ignore-all these are all devDependencies of the Multichain Test Dapp that are NOT part of the build pipeline.

jiexi committed:
lavamoat

jiexi committed:
Add test-dapp-multichain to depcheckrc and yarn command

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
fix build system policy

jiexi committed:
yarn

jiexi committed:
lavamoat part2

jiexi committed:
Fix secp256k1@5.0.0 audit vuln

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
yarn dedupe

jiexi commented:
@metamaskbot update-policies

jiexi committed:
Fix mask-type

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

metamaskbot committed:
Update LavaMoat policies

jiexi committed:
yarn dedupe

jiexi committed:
Merge remote-tracking branch 'origin/ad/multichain-api-e2e-test' into ad/multichain-api-e2e-test

jiexi committed:
Trigger

jiexi committed:
restore lavamoat

jiexi committed:
lavamoat

adonesky1 committed:
lavamoat

jiexi committed:
use test-dapp-multichain 0.4.1

jiexi committed:
fix lavamoat build policy

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
dedupe

jiexi commented:
@metamaskbot update-policies

jiexi committed:
fix snapshot

metamaskbot committed:
Update LavaMoat policies

jiexi committed:
Merge remote-tracking branch 'origin/ad/multichain-api-e2e-test' into ad/multichain-api-e2e-test

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

adonesky1 committed:
add back requestPermissionApprovalForOrigin

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
Merge remote-tracking branch 'origin/ad/multichain-api-e2e-test' into ad/multichain-api-e2e-test

adonesky1 committed:
fix expected hooks issues

jiexi committed:
fix snapshot

jiexi committed:
Fix all the snapshots instead of one at a time lol

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
fix snapshot

ffmcgee725 committed:
e2e testing: wallet_getSession - a dapp can check its existing permissions (#29298)

Adding e2e test cases for [a dapp can checking its existing
permissions](https://app.zenhub.com/workspaces/wallet-api-platform-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/3815)

ffmcgee725 committed:
e2e testing: unsupported networks (uninstalled EVM networks) requested as part of `wallet_createSession` request#3817 (#29326)

wallet_createSession - wallet will ignore unsupported networks [#3817](https://app.zenhub.com/workspaces/wallet-api-platform-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/3817)

adonesky1 committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

adonesky1 commented:
@metamaskbot update-policies

adonesky1 committed:
fix build system policy file

adonesky1 commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

adonesky1 committed:
lint

adonesky1 committed:
more lint

adonesky1 committed:
one more lint

adonesky1 committed:
fix build system policy file?

adonesky1 commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

adonesky1 commented:
@metamaskbot update-policies

adonesky1 committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

ffmcgee725 committed:
e2e testing: accounts requested in `wallet_createSession`#3822 (#29356)

test: add [e2e test for accounts requested in
wallet_createSession](https://app.zenhub.com/workspaces/wallet-api-platform-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/3822)

jiexi commented:
@metamaskbot update-policies

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

# Conflicts:
#	lavamoat/browserify/beta/policy.json
#	lavamoat/browserify/flask/policy.json
#	lavamoat/browserify/main/policy.json
#	lavamoat/browserify/mmi/policy.json
#	lavamoat/build-system/policy.json
#	package.json
#	ui/pages/confirmations/components/confirm/info/approve/approve-details/__snapshots__/approve-details.test.tsx.snap
#	ui/pages/confirmations/components/confirm/info/shared/send-heading/__snapshots__/send-heading.test.tsx.snap
#	ui/pages/confirmations/components/confirm/info/token-transfer/__snapshots__/token-transfer.test.tsx.snap
#	yarn.lock

jiexi commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

ffmcgee725 committed:
e2e testing: user can edit the EVM account permissions in `wallet_createSession` request#3827 (#29400)

-- test: e2e test for user can edit the EVM account permissions in wallet_createSession request

jiexi committed:
yarn dedupe

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
Merge remote-tracking branch 'origin/ad/multichain-api-e2e-test' into ad/multichain-api-e2e-test

jiexi committed:
lint

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
lavamoat

jiexi committed:
snapshot

jiexi commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

adonesky1 commented:
@metamaskbot update-policies

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
Merge remote-tracking branch 'origin/ad/multichain-api-e2e-test' into ad/multichain-api-e2e-test

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi commented:
@metamaskbot update-policies

jiexi committed:
Merge branch 'jl/caip-multichain-migrate-core' into ad/multichain-api-e2e-test

jiexi committed:
Jl/multichain api e2e test refactor (#29464)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

* Replaces real scopes with localhost scopes when actual connections are
being made
* Refactor edit network form helper
* Refactor edit account form helper

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/29464?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.
